ServerCore/Pages/Teams/Play.cshtml.cs

```diff
@@ -64,41 +68,55 @@ public async Task OnGetAsync(SortOrder? sort, int teamId)
                                   join PuzzleStatePerTeam pspt in stateForTeamQ on puzzle.ID equals pspt.PuzzleID
-                                  select new { Puzzle = puzzle, State = pspt };
+                                  select new PuzzleView { ID = puzzle.ID, Group = puzzle.Group, OrderInGroup = puzzle.OrderInGroup, Name = puzzle.Name, CustomUrl = puzzle.CustomURL, SolvedTime = pspt.SolvedTime };
 
             switch (sort ?? DefaultSort)
             {
                 case SortOrder.PuzzleAscending:
-                    visiblePuzzlesQ = visiblePuzzlesQ.OrderBy(puzzleWithState => puzzleWithState.Puzzle.Name);
+                    visiblePuzzlesQ = visiblePuzzlesQ.OrderBy(pv => pv.Name);
                     break;
                 case SortOrder.PuzzleDescending:
-                    visiblePuzzlesQ = visiblePuzzlesQ.OrderByDescending(puzzleWithState => puzzleWithState.Puzzle.Name);
+                    visiblePuzzlesQ = visiblePuzzlesQ.OrderByDescending(pv => pv.Name);
                     break;
                 case SortOrder.GroupAscending:
-                    visiblePuzzlesQ = visiblePuzzlesQ.OrderBy(puzzleWithState => puzzleWithState.Puzzle.Group).ThenBy(puzzleWithState => puzzleWithState.Puzzle.OrderInGroup);
+                    visiblePuzzlesQ = visiblePuzzlesQ.OrderBy(pv => pv.Group).ThenBy(pv => pv.OrderInGroup);
                     break;
                 case SortOrder.GroupDescending:
-                    visiblePuzzlesQ = visiblePuzzlesQ.OrderByDescending(puzzleWithState => puzzleWithState.Puzzle.Group).ThenByDescending(puzzleWithState => puzzleWithState.Puzzle.OrderInGroup);
+                    visiblePuzzlesQ = visiblePuzzlesQ.OrderByDescending(pv => pv.Group).ThenByDescending(pv => pv.OrderInGroup);
                     break;
                 case SortOrder.SolveAscending:
-                    visiblePuzzlesQ = visiblePuzzlesQ.OrderBy(puzzleWithState => puzzleWithState.State.SolvedTime ?? DateTime.MaxValue);
+                    visiblePuzzlesQ = visiblePuzzlesQ.OrderBy(pv => pv.SolvedTime ?? DateTime.MaxValue);
                     break;
                 case SortOrder.SolveDescending:
-                    visiblePuzzlesQ = visiblePuzzlesQ.OrderByDescending(puzzleWithState => puzzleWithState.State.SolvedTime ?? DateTime.MaxValue);
+                    visiblePuzzlesQ = visiblePuzzlesQ.OrderByDescending(pv => pv.SolvedTime ?? DateTime.MaxValue);
                     break;
                 default:
                     throw new ArgumentException($"unknown sort: {sort}");
             }
 
+            PuzzleViews = await visiblePuzzlesQ.ToListAsync();
+
             Dictionary<int, ContentFile> files = await (from file in _context.ContentFiles
                                                         where file.Event == Event && file.FileType == ContentFileType.Puzzle
                                                         select file).ToDictionaryAsync(file => file.Puzzle.ID);
```

Comment:
My guess is that this line is the real culprit for over-querying -- I think file.Puzzle is triggering a query for every file

Reply:
OHHH makes sense.

Reply:
@morganbr , can I just expose ContentFile.PuzzleID? It's commented out with guidance I don't understand. ContentFile already exposes EventID...

Reply:
You can give it a shot. I seem to recall it might have had something to do with getting funny migrations, so please make sure adding a migration is a no-op.

Reply:
namespace Data.Migrations
{
    public partial class PuzzleID : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {

        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {

        }
    }
}